### PR TITLE
Implement basic map

### PR DESCRIPTION
**Pull request by TPGamesNL**
Implements the basic map, currently doesn't have an overlay for music memory, I'll add that (in a subclass) when we have a system for getting music memories.

There are still some TODOs left:
- Properly handle permissions, probably in a utility class for them. Android has some strict guidelines for them.
- When opening the map, start at the user's location instead of a view of the Netherlands

### CODE REVIEW

**Comment by TPGamesNL** [comment]
Also apparently CI takes 2-3 minutes longer than before cause of osmdroid now

**Comment by andreitudorp** [comment]
> Also apparently CI takes 2-3 minutes longer than before cause of osmdroid now

I think after it checks it once it should not check it anymore if no changes were made. Right?

**Comment by TPGamesNL** [comment]
Yea it only checks when you push to a PR branch or when you merge a PR

File: app/src/main/java/com/example/musicmap/screens/map/MapFragment.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+            // Request coarse location access permission
+            activity.requestPermissions(new String[] {Manifest.permission.ACCESS_COARSE_LOCATION},
+                    PERMISSION_ACCESS_COARSE_LOCATION_REQUEST_CODE);
+        }
+        if (activity.checkSelfPermission(Manifest.permission.ACCESS_FINE_LOCATION)
+                != PackageManager.PERMISSION_GRANTED) {
+            // Request fine location access permission
+            activity.requestPermissions(new String[] {Manifest.permission.ACCESS_FINE_LOCATION},
+                    PERMISSION_ACCESS_FINE_LOCATION_REQUEST_CODE);
+        }
```
[the remark this answers is not in the document]
Comment (TPGamesNL) [reply app/src/main/java/com/example/musicmap/screens/map/MapFragment.java:86]: Yes, I will do this when I do permissions properly in another class

File: app/src/main/java/com/example/musicmap/screens/map/MapFragment.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+        @Override
+        protected void drawMyLocation(Canvas canvas, Projection pj, Location lastFix) {
+            lastFix.removeBearing();
+
+            super.drawMyLocation(canvas, pj, lastFix);
+        }
+    }
+
+}
```
[the remark this answers is not in the document]
Comment (TPGamesNL) [reply app/src/main/java/com/example/musicmap/screens/map/MapFragment.java:194]: It is implemented now

File: app/src/main/java/com/example/musicmap/screens/map/MapFragment.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+     *
+     * @param overlay the overlay to add.
+     */
+    protected void addOverlay(Overlay overlay) {
+        OverlayManager overlayManager = mapView.getOverlayManager();
+        overlayManager.add(overlayManager.size(), overlay);
+    }
+
+    /**
+     * A map overlay showing your current location, but without bearing.
```
[the remark this answers is not in the document]
Comment (TPGamesNL) [reply app/src/main/java/com/example/musicmap/screens/map/MapFragment.java:174]: Probably; I'll do this when I add more overlays in the future (music memories and stuff)

File: app/src/main/java/com/example/musicmap/util/ui/FragmentUtil.java [hunk 4]
```diff
@@ -37,6 +37,7 @@ public static void initFragment(@NonNull FragmentManager fragmentManager,
     public static void replaceFragment(@NonNull FragmentManager fragmentManager,
                                        @IdRes int containerID,
                                        @NonNull Class<? extends Fragment> fragmentClass) {
+        // TODO don't replace if that fragment in the container
```
[the remark this answers is not in the document]
Comment (TPGamesNL) [reply app/src/main/java/com/example/musicmap/util/ui/FragmentUtil.java]: Fixed

**Review by andreitudorp — APPROVED** [review]
Good job! We can merge this changes!

**Pull request merged**